core/kernel/shared/require.rb

```diff
@@ -774,3 +774,24 @@
     }
   end
 end
+
+ruby_version_is "2.7" do
+  describe "$LOAD_PATH.resolve_feature_path" do
+    it "returns what will be loaded without actual loading, .rb file" do
+      extension, path = $LOAD_PATH.resolve_feature_path('set')
+      extension.should == :rb
+      path.should.end_with?('/set.rb')
+    end
+
+    it "returns what will be loaded without actual loading, .so file" do
+      extension, path = $LOAD_PATH.resolve_feature_path('etc')
+      extension.should == :so
+      path.should =~ /\/etc\.(so|bundle)$/
```

Comment:
Could you use `.end_with?("etc.#{RbConfig::CONFIG['DLEXT']}")`? (and add `require 'rbconfig` at the start of the `it`)